scripts/us_cdc/nndss_weekly_tables/process_weekly_data.py

```diff
@@ -0,0 +1,616 @@
+                        'Reporting Area': [place],
+                        'column_name': [column],
+                        'variableMeasured': [dcid]
+                    })
+
+                    dcid_df = pd.concat([dcid_df, concat_df])
+
+                else:
+                    # get name and dcid
+                    svdict_tmp = svdict.copy()
+                    dcid = get_statvar_dcid(svdict_tmp)
+                    # remove the logical Or from dcid
+                    dcid = dcid.replace('Or', '_')
+                    # remove AllSerotypes and AllSerogroups from dcid
+                    dcid = dcid.replace('_AllSerogroups', '')
+                    dcid = dcid.replace('_AllSerotypes', '')
+
+                    svdict_tmp['name'] = make_name_str(column_components, match,
+                                                       dcid)
+                    # add to column_map
+                    key = 'Node: dcid:' + dcid
+                    if key not in sv_map.keys():
+                        sv_map[key] = svdict_tmp
+                    concat_df = pd.DataFrame({
+                        'Reporting Area': [place],
+                        'column_name': [column],
+                        'variableMeasured': [dcid]
+                    })
+
+                    dcid_df = pd.concat([dcid_df, concat_df])
+
+    return sv_map, dcid_df
+
+
+def generate_processed_csv(input_path):
+    processed_dataframe_list = []
+    #TODO: Parallelize this step
+    # for all files in the data directory process the files and append to a common dataframe
+    for filename in sorted(os.listdir(input_path)):
+
+        table_id = filename.split('table_')[1].split('.')[0]
+        lowercase_in_tablename = re.findall(r'[a-z]', table_id)
+        if filename.endswith('.csv') and len(lowercase_in_tablename) < 2:
+            file_inputpath = os.path.join(input_path, filename)
+            df = process_nndss_current_weekly(file_inputpath)
+            processed_dataframe_list.append(df)
+    # concat list of all processed data frames
+
+    cleaned_dataframe = pd.concat(processed_dataframe_list, ignore_index=False)
+    logging.info(
+        f"Count of processed_dataframe_list {len(processed_dataframe_list)}")
+    del processed_dataframe_list
+
+    # set measurement method
+    cleaned_dataframe[
+        'measurementMethod'] = 'dcs:CDC_NNDSS_Diseases_WeeklyTables'
+
+    cleaned_dataframe['value'] = cleaned_dataframe.apply(
+        lambda x: float(x['value']) if len(x['value']) > 0 else np.nan, axis=1)
+    cleaned_dataframe.dropna(subset=['value', 'observationAbout'], inplace=True)
+    cleaned_dataframe['value'] = cleaned_dataframe['value'].astype(float)
+    cleaned_dataframe.to_csv("processed_raw_weekly_data.csv", index=False)
+
+    return cleaned_dataframe
+
+
+def main(_) -> None:
+    mode = FLAGS.mode
+    if mode == "" or mode == "download":
+        _START = FLAGS.start_year
+        _END = FLAGS.end_year
+        flags.DEFINE_string(
```

Comment:
Define this at the beginning of the code, along with other flags

Reply:
updated